docs-src/typescript/how-to-replay-a-workflow-execution-in-typescript.md

```diff
@@ -9,14 +9,38 @@ tags:
   - typescript
 ---
 
-Retrieve Event History with [`WorkflowService.getWorkflowExecutionHistory`](https://typescript.temporal.io/api/classes/proto.temporal.api.workflowservice.v1.workflowservice-1/#getworkflowexecutionhistory).
+Use [worker.runReplayHistories](https://typescript.temporal.io/api/classes/worker.Worker#runreplayhistories)
+or [worker.runReplayHistory](https://typescript.temporal.io/api/classes/worker.Worker#runreplayhistory)
+to replay multiple or one Workflow Histories.
 
-<!--SNIPSTART typescript-history-get-workflowhistory-->
-<!--SNIPEND-->
+In the following example (which requires advanced visibility to be enabled), histories are
+downloaded from the server and then replayed by passing in a client and a set of executions:
 
-Then call [`Worker.runReplayHistory`](https://typescript.temporal.io/api/classes/worker.worker/#runreplayhistory).
+```ts
+const executions = client.workflow.list({ query: 'TaskQueue=foo` });
```

Comment:
Few things:
1. the query should set a limit, probably using a date range
2. we should mention that as of server version 1.18 this query is only supported with ES based visibility